subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_pdu.h

```diff
@@ -4,6 +4,16 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#if defined(CONFIG_BT_CTLR_ADV_PDU_LINK)
+#define PDU_ADV_MEM_SIZE       MROUND(PDU_AC_LL_HEADER_SIZE + \
+				      PDU_AC_PAYLOAD_SIZE_MAX + 4)
+#define PDU_ADV_NEXT_PTR(p)    *(struct pdu_adv **)((uint8_t *)(p) + \
+						    PDU_ADV_MEM_SIZE - 4)
```

Comment:
This places pointer at the very end of PDU buffer which is convenient for two reasons:
- it will not be overwritten when updating PDU contents
- it can be calculated by compiler
The slight disadvantage is that MROUND macro is used here so anything that includes this file needs to include util/mem.h as well.

Another option would be to place pointer just after PDU data - it can be easily calculated at runtime, but may be corrupted when PDU is modified so need to take extra care.

Placing pointer at the beginning of buffer is not a good option since it requires extra handling for PDU data which can start on different offset depending on configuration.

Reply:
will change to `sizeof(uintptr_t)`